src/main/java/seedu/duke/parsermanager/ParseAddClient.java

```diff
@@ -0,0 +1,196 @@
+package seedu.duke.parsermanager;
+
+import seedu.duke.command.Command;
+import seedu.duke.command.CommandAddClient;
+import seedu.duke.exception.EmptyDetailException;
+import seedu.duke.exception.IncorrectFlagOrderException;
+import seedu.duke.exception.InvalidBudgetFormatException;
+import seedu.duke.exception.InvalidContactNumberException;
+import seedu.duke.exception.InvalidEmailException;
+import seedu.duke.exception.MissingFlagException;
+
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static seedu.duke.CommandStructure.ADD_CLIENT_FLAGS;
+import static seedu.duke.Messages.EXCEPTION;
+import static seedu.duke.Messages.MESSAGE_ADD_CLIENT_WRONG_FORMAT;
+import static seedu.duke.Messages.MESSAGE_INVALID_BUDGET_FORMAT;
+import static seedu.duke.Messages.MESSAGE_INVALID_CONTACT_NUMBER;
+import static seedu.duke.Messages.MESSAGE_INVALID_EMAIL;
+
+public class ParseAddClient extends Parser {
+    private final String commandDescription;
+
+    public ParseAddClient(String addCommandDescription) {
+        this.commandDescription = addCommandDescription;
+    }
+
+
+    public Command parseCommand() throws MissingFlagException, IncorrectFlagOrderException, EmptyDetailException,
+            InvalidContactNumberException, InvalidEmailException, InvalidBudgetFormatException {
+        try {
+            checkForEmptyDetails(commandDescription);
+
+            ArrayList<String> clientDetails = processCommandAddClientDetails(commandDescription);
+            validateClientDetails(clientDetails);
+            return new CommandAddClient(clientDetails);
+
+        } catch (MissingFlagException e) {
+            throw new MissingFlagException(MESSAGE_ADD_CLIENT_WRONG_FORMAT);
+        } catch (IncorrectFlagOrderException e) {
+            throw new IncorrectFlagOrderException(MESSAGE_ADD_CLIENT_WRONG_FORMAT);
+        } catch (InvalidContactNumberException e) {
+            throw new InvalidContactNumberException(MESSAGE_INVALID_CONTACT_NUMBER);
+        } catch (EmptyDetailException e) {
+            throw new EmptyDetailException(MESSAGE_ADD_CLIENT_WRONG_FORMAT);
+        } catch (InvalidEmailException e) {
+            throw new InvalidEmailException(MESSAGE_INVALID_EMAIL);
+        } catch (InvalidBudgetFormatException e) {
+            throw new InvalidBudgetFormatException(MESSAGE_INVALID_BUDGET_FORMAT);
+        }
+    }
+
+    private void checkForEmptyDetails(String commandDetail) throws EmptyDetailException {
```

Comment:
These methods are common across all parse classes, can put these common methods in parse superclass and let the subclasses call them via super